pyomo/repn/plugins/baron_writer.py

```diff
@@ -173,6 +173,14 @@ def _monomial_to_string(self, node):
         return ftoa(const, True) + '*' + self.smap.getSymbol(var)
 
     def _linear_to_string(self, node):
+        values = [
+            self._monomial_to_string(arg)
+            if (arg.__class__ is EXPR.MonomialTermExpression
+                and not arg.arg(1).is_fixed())
+            else ftoa(arg)
+            for arg in node.args
+        ]
+        return node._to_string(values, False, self.smap)
```

Comment:
It looks like the following lines are no longer needed as this method will return here.